plugins/utils/which-key.nix

```diff
@@ -484,33 +490,46 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin config {
     delay = 200;
 
     spec = [
```

Comment:
We could do `spec = options.plugins.which-key.settings.spec.example` so we don't need to come up with two examples?

Reply:
Actually, I think we'd have to use `(options.plugins.which-key.settings.type.getSubOptions [ ]).spec.example`, so maybe it's more readable to define the example in a let block and use it in both instances, or use a `rec` attrset and reference the local definition recursively, e.g. use `rec` to assign `example = settingsExample.spec` in the spec option declaration.

Or just do it how you're doing it already; i.e. have two examples.

Reply:
Moved them to a shared let block so we can manage them in one place.